srcalc/src/test/java/gov/va/med/srcalc/vista/RpcVistaPatientDaoTest.java

```diff
@@ -157,4 +160,36 @@ public final void testLabResponseNoUnits()
         // Insure the lab was not added and no exceptions occurred.
         assertEquals(1, patient.getLabs().size());
     }
+    
+    @Test
+    public final void testHealthFactorsValid()
+    {
+        final VistaProcedureCaller caller = mockVistaProcedureCaller();
+        when(caller.doRpc(
+                RADIOLOGIST_DUZ,
+                RemoteProcedure.GET_HEALTH_FACTORS,
+                String.valueOf(PATIENT_DFN)))
+                .thenReturn(Arrays.asList(VALID_HEALTH_FACTORS.split("\n")));
+        final RpcVistaPatientDao dao = new RpcVistaPatientDao(caller, RADIOLOGIST_DUZ);
+        final Patient patient = dao.getPatient(PATIENT_DFN);
+        assertEquals(3, patient.getHealthFactors().size());
+        // Checking all of the items so that we can verify proper filtering.
+        assertEquals("08/22/2014 DEPRESSION ASSESS POSITIVE (MDD)", patient.getHealthFactors().get(0));
+        assertEquals("08/20/2014 ALCOHOL - TREATMENT REFERRAL", patient.getHealthFactors().get(1));
+        assertEquals("07/30/2014 GEC HOMELESS", patient.getHealthFactors().get(2));
+    }
+    
+    @Test
+    public final void testNoHealthFactors()
+    {
+        final VistaProcedureCaller caller = mockVistaProcedureCaller();
+        when(caller.doRpc(
+                RADIOLOGIST_DUZ,
+                RemoteProcedure.GET_HEALTH_FACTORS,
+                String.valueOf(PATIENT_DFN)))
+                .thenReturn(new ArrayList<String>());
+        final RpcVistaPatientDao dao = new RpcVistaPatientDao(caller, RADIOLOGIST_DUZ);
+        final Patient patient = dao.getPatient(PATIENT_DFN);
+        assertEquals(0, patient.getHealthFactors().size());
```

Comment:
Again, just compare against an empty list. It doesn't save any lines here but it would result in a better failure message.